shell/common/shell.cc

```diff
@@ -380,6 +380,17 @@ void Shell::NotifyLowMemoryWarning() const {
   // to purge them.
 }
 
+Engine::RunStatus Shell::RunEngine(RunConfiguration run_configuration) {
```

Comment:
The expectation is that all shell methods are called on the platform task runner. So let's post the task to the correct task runner. I don't think anyone ever uses the run status to log an error. So, we can return void here. Or, better still, take a callback that returns the result.

Reply:
Fixed.  I've also "fixed" the problems I was having with Flutter tester by just exposing the things it needs through the shell.  We may want to remove those but that will be an effort I'm less familiar with the internals of at this point.

Additionally, I'm adding back the getEngine but marking ti deprecated to make the eventual roll to Fuchsia less painful.